src/components/Tag/index.tsx

```diff
@@ -1,115 +1,103 @@
+const StyledContainer = styled('span', {
+  shouldForwardProp: prop => !['variant'].includes(prop.toString()),
+})<{ variant: Color | 'disabled' }>`
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  white-space: nowrap;
+  border-radius: ${({ theme }) => theme.space['0.5']};
+  padding: 0 ${({ theme }) => theme.space['1']};
+  gap: ${({ theme }) => theme.space['1']};
+  width: fit-content;
+  height: ${({ theme }) => theme.space['3']};
+  ${({ variant, theme }) => {
+    if (variant === 'disabled')
+      return `
+      color: ${theme.colors.neutral.textDisabled};
+      background: ${theme.colors.neutral.backgroundDisabled};
+      border: solid 1px ${theme.colors.neutral.borderDisabled};
+      cursor: not-allowed
 
-type StyledContainerProps = {
-  disabled: boolean
-  variant: TagVariant
-} & BoxProps
+    `
+    if (variant === 'neutral')
+      return `
+      color: ${theme.colors.neutral.text};
+      background: ${theme.colors.neutral.background};
+      border: solid 1px ${theme.colors.neutral.border};
+    `
 
-const StyledContainer = styled(Box, {
-  shouldForwardProp: props =>
-    !['disabled', 'variant'].includes(props.toString()),
-})<StyledContainerProps>`
-  ${({ disabled }) => disabled && disabledStyles}
-  border-radius: 4px;
-  justify-content: center;
-  display: flex;
-  ${variantStyles}
+    return `
+      color: ${theme.colors[variant].text};
+      background: ${theme.colors[variant].background};
+      border: solid 1px ${theme.colors[variant].borderDisabled};
+    `
+  }}
 `
 
-const StyledText = styled('span')`
-  ${({ 'aria-disabled': disabled }) => disabled && disabledStyles}
-  color: ${({ theme }) => theme.colors.neutral.text};
-  font-size: 14px;
-  align-self: center;
-  max-width: 350px;
-  overflow: hidden;
-  white-space: pre;
-  text-overflow: ellipsis;
+const StyledTypography = styled(Typography)`
+  font-size: 12px;
+  font-weight: 500;
+  color: inherit;
 `
 
 const StyledTouchable = styled(Touchable, {
   shouldForwardProp: props => !['variant'].includes(props.toString()),
-})<{ variant: TagVariant }>`
-  margin-left: 16px;
+})<{ variant: Color }>`
   display: flex;
   align-items: center;
-  justify-content: center;
-  &:hover,
-  &:focus {
-    background-color: ${({ theme }) => theme.colors.primary.backgroundHover};
+  svg {
+    fill: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+  &:hover:enabled,
+  &:focus:enabled {
     svg {
-      fill: ${({ theme }) => theme.colors.primary.text};
+      fill: ${({ theme }) => theme.colors.neutral.textHover};
     }
   }
-  ${({ variant, theme }) =>
-    variant === 'bordered' &&
-    `
-  border-radius: 4px;
-  border: 1px solid ${theme.colors.neutral.borderWeak};
-  padding: 4px;
-  width: 32px;
-  height: 32px;
-`}
 `
 
 type TagProps = {
-  children: ReactNode
-  disabled?: boolean
   isLoading?: boolean
   onClose?: MouseEventHandler<HTMLButtonElement>
   textStyle?: JSX.IntrinsicAttributes['css']
```

Comment:
what you mean by complex children ?

Reply:
we can't pass anything else than text here, which is styled depending on variant.

for `TextStyle` It's something we use in shire to add `maxwidth` and `overflow hidden` with tooltip to avoid very big text content.

what we could do is use Tooltip and max width in the tag it self i will see with design team